dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/report/MetadataReportFactory.java

```diff
@@ -34,4 +34,9 @@ public interface MetadataReportFactory {
     MetadataReport getMetadataReport(URL url);
 
     default void destroy() {}
+
+    @Adaptive({PROTOCOL_KEY})
+    default String relatedRegistryIdSuffix(URL url) {
+        return "";
+    }
```

Comment:
This method here makes the interface definition hard to understand.

Factory is only responsible for creating new instances.